api/service/syncing/syncing.go

```diff
@@ -106,6 +107,26 @@ type StateSync struct {
 	syncMux            sync.Mutex
 }
 
+// level = 0 only clean "old" blocks, level > 0 clean up everything
+func (ss *StateSync) cleanCache(level int) {
```

Comment:
Code smell is strong with this one.

1. Lines 119–127 are an almost exact duplicate of lines 114–118, with only one diffecence: `configPeer.newBlocks` is `nil`-ed out.
2. `level int` arguments looks like a magic constant from call sites (e.g. around line 683, where we call this twice, once with 0 then again with 1.  Reader of that code will not know what is going on.

Can you please refactor this?  After refactoring, the following should be met:

1. `cleanCache(0)` ⇒ `purgeOldBlocksFromCache()`
2. `cleanCache(1)` ⇒ `purgeAllBlocksFromCache()`
3. Looping is done only once.
4. If you need a `clear all` boolean flag, the flag should be of type `bool` and named appropriately, e.g. `purgeNewOnesToo bool`.

Reply:
refactor the cleanCache into two for readability